### ISSUE TITLE
ruby versions - need to migrate support 

### ISSUE THREAD

**Issue by yannickwurm** [wurmlab/sequenceserver#585]
So ruby 2.6.10 is no longer LTS supported (i.e. no more security updates). 

Some recent installs using ruby 3 have highlighted issues. Probably we will want to get everything to work with newer ruby (e.g., [3.1.2](https://rubychangelog.com/versions-all/#ruby-312))

https://endoflife.date/ruby

**Comment by lukaszsobala** [comment]
Sequenceserver does not work for me with ruby 3.0 (on update from ubuntu 21.10 to 22.04, ruby 2,7 is no longer available from the main repository). Upon invoking `sequenceserver`, the error is:
```
<internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- webrick (LoadError)
	from <internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /var/lib/gems/3.0.0/gems/rack-2.2.3/lib/rack/handler/webrick.rb:3:in `<top (required)>'
	from <internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /var/lib/gems/3.0.0/gems/sequenceserver-2.0.0/lib/sequenceserver/server.rb:1:in `<top (required)>'
	from <internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /var/lib/gems/3.0.0/gems/sequenceserver-2.0.0/lib/sequenceserver.rb:21:in `<module:SequenceServer>'
	from /var/lib/gems/3.0.0/gems/sequenceserver-2.0.0/lib/sequenceserver.rb:6:in `<top (required)>'
	from <internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/usr/lib/ruby/vendor_ruby/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /var/lib/gems/3.0.0/gems/sequenceserver-2.0.0/bin/sequenceserver:177:in `block (2 levels) in <top (required)>'
	from /var/lib/gems/3.0.0/gems/slop-3.6.0/lib/slop.rb:260:in `parse!'
	from /var/lib/gems/3.0.0/gems/slop-3.6.0/lib/slop.rb:65:in `parse!'
	from /var/lib/gems/3.0.0/gems/sequenceserver-2.0.0/bin/sequenceserver:48:in `<top (required)>'
	from /usr/local/bin/sequenceserver:25:in `load'
	from /usr/local/bin/sequenceserver:25:in `<main>'
```
With 2.7 it worked mostly OK.

Cheers

**Comment by yeban** [comment]
@lukaszsobala This is being worked on in PR #590. Maybe give @iQuxLE's branch a try and help us be sure of the changes? He is working on a Mac. Your testing will help us be sure of the changes for Ubuntu.

**Comment by iQuxLE** [comment]
@lukaszsobala 

Does it work for you now? Your error was mainly because in Ruby 3.0 webrick is no more bundled and you have to gem install or include it in dependencies of `squenceserver.gemspec`

**Comment by lukaszsobala** [comment]
@yeban @iQuxLE I would love to give it a try, but I don't really know how to "try a branch"... Until now I just used gem to install sequenceserver (and nothing else really), and I don't know what bundler does. Could you explain how to do this?

**Comment by lukaszsobala** [comment]
@iQuxLE thanks. Now when I do:
```
git clone https://github.com/iQuxLE/sequenceserver.git
cd sequenceserver
git checkout abf47f5ea6f34caff541b986b3366184230908e1
sudo bundle install --without=development
```

I get the following error:
`sequenceserver-2.0.0 requires ruby version >= 3.1.2, which is incompatible with the current version, ruby 3.0.2p107`

Which version of ruby, then, should I use? I can try replacing the system version with one installed through mamba for example, but in this case I could also go back to 2.7... It seems to be a problem with Ubuntu repositories but then again, people installing sequenceserver for the first time on an LTS release would find it confusing.

**Comment by iQuxLE** [comment]
You can change `sequenceserver.gemspec` (last file) to Ruby version : `>= 2.5` or `= 3.0`
It should work for you after that

**Comment by lukaszsobala** [comment]
> You can change `sequenceserver.gemspec` (last file) to Ruby version : `>= 2.5` or `= 3.0` It should work for you after that

Indeed @iQuxLE! now it works without a problem. Thank you.

**Comment by yannickwurm** [comment]
Closed with [25f6a26](https://github.com/wurmlab/sequenceserver/commit/25f6a26baed54f6bc6fcf79ee429f33e39877327) - thank you @tadast & @iQuxLE
